### PR TITLE
Update routes approche

### PR DESCRIPTION
**Pull request by ichala**
I'm Opening This Pr because the approach that we followed me and @d2ndjim to map through our routes looks so complex and feels wrong, so I did some research and came up with a solution using [UserRoutes](https://reactrouter.com/en/6.4.0/hooks/use-routes) method from react-routes-dom which help a lot in organizing our code , I created a component that checks if user logged in or no and all of the private routes in our routes config files will be rendered as children

```JS
import React from 'react';
import { useSelector } from 'react-redux';
import { Navigate, Outlet } from 'react-router-dom';

function PrivateRoutes() {
  const user = useSelector((state) => state.user);
  if (!user) {
    return (
      <Navigate to="/login" replace />
    );
  }
  return <Outlet />;
}

export default PrivateRoutes;

```

this approach will make the code more clean and organized

### CODE REVIEW

File: src/components/Auth/PrivateRoutes.jsx [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  const user = useSelector((state) => state.user);
+  if (!user) {
+    return (
+      <Navigate to="/login" replace />
+    );
+  }
+  return <Outlet />;
+}
+
+export default PrivateRoutes;
```
Comment (LYANGEND) [inline src/components/Auth/PrivateRoutes.jsx:15]: Good job implementing the PrivateRoutes function to render private routes. I can also suggest that the PrivateRoutes component deals only with authenticated users

**Review by LYANGEND — APPROVED** [review]
<h1>Status: Approve 🥇 ✔</h1>
Hello @ichala , 

### In my opinion your task is complete!  There is nothing else to say other than....good work 👍  
## Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

**Review by aimalamiri — APPROVED** [review]
### **Hi 👋** 

### **Congratulations! 🎉**

Your project is complete!  There is nothing else to say other than... it's time to merge it 🚀 : 
 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**Review by EidHachem — APPROVED** [review]
Hello @ichala,

Well done, approved by me ✔️
it's time to merge 🎉

**Pull request merged**